controllers/storagecluster/storageclasses.go

```diff
@@ -217,7 +218,8 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 			if existing.DeletionTimestamp != nil {
 				return fmt.Errorf("failed to restore StorageClass  %s because it is marked for deletion", existing.Name)
 			}
-			if !reflect.DeepEqual(sc.Parameters, existing.Parameters) {
+			if !reflect.DeepEqual(sc.Parameters, existing.Parameters) ||
+				!reflect.DeepEqual(sc.Annotations, existing.Annotations) {
```

Comment:
Checking if this creates the problem when we add annotation on the SC like for DR or not (which am not sure), @ShyamsundarR @umangachapagain will there be any problem if we override the annotations on the SC?

Reply:
As long as we merge all the annotations and not remove it, we should be fine.

Reply:
Recreating (or updating) the StorageClass should preserve existing annotations and labels, so if the change is overwriting these then we would lose information, which should not happen. (I have not looked at code, answering based on the comment here)

Reply:
@nb-ohad I recall you saying somewhere that the SCs created by the ocs-operator are managed and user is not supposed to make edits to it. Would like to know your thoughts on this?

Reply:
The spec section and name (or related metadata) should be retained by the operator. User additions like annotations and labels (and even finalizers) should not be overwritten.

Reply:
In cases where the params do not differ, I have updated the PR to update the annotations instead of recreating the SC.

cc: @Madhu-1